### ISSUE TITLE
Map.merge does not return this? withMutations

### ISSUE THREAD

**Issue by mailaneel** [facebook/immutable-js#350]
This does not work

```javascript
var a,b,
a = new Immutable.Map();
b = a.withMutations(function(_a){
   
// below code does not return this 
 _a.merge({x: 1});
});

console.log(b.get('x')); // undefined
```

below code works fine

```javascript
var a,b,
a = new Immutable.Map({x:1});
b = a.withMutations(function(_a){   
 _a.merge({y: 1});
});

console.log(b.toJSON()); // {x:1, y:1}
```

**Comment by rtsao** [comment]
+1

**Comment by hufeng** [comment]
Yes,  cursor has the same problem.

```javascript

var Immutable = require('immutable');
var Cursor = require('immutable/contrib/cursor');


var a = Immutable.Map();
var c = Cursor.from(a, function onChange(next) {console.log(next)});
var b = c.withMutations(function(_c) {return _c.merge({x: 1})});
console.log(b);


var a1 = Immutable.Map({});
var c1 = Cursor.from(a1, function onChange(next) {console.log(next)});
var b1 = c1.withMutations(function(_c1) { return _c1.merge({x:1})});
console.log(b1);

var a2 = Immutable.Map({x: 1});
var c2 = Cursor.from(a2, function onChange(next) {console.log(next)});
var b2 = c2.withMutations(function(_c2) { return _c2.merge({y:1})});
console.log(b2);


//output
Map {}
Cursor {}

Map {}
Cursor {}

Map { x: 1, y: 1 }
Cursor { x: 1, y: 1 }
```

**Comment by leebyron** [comment]
This should be partially solved by #507
